modules/caas/backend/src/main/java/io/cattle/platform/inator/unit/PortUnit.java

```diff
@@ -117,8 +117,9 @@ public void bind(InatorContext context, Map<String, Object> instanceData) {
                 if (spec.getPublicPort() == null && spec.getPrivatePort() == port) {
                     Object owner = getOwner(getDeploymentUnit(context));
                     Account account = getAccount(owner);
-                    PooledResource resource = svc.lockManager.lock(new PortUnitLock(account, this), () -> {
-                        return svc.poolManager.allocateOneResource(account, owner,
+                    Account clusterAccount = svc.clusterDao.getOwnerAcccountForCluster(account.getClusterId());
+                    PooledResource resource = svc.lockManager.lock(new PortUnitLock(clusterAccount, this), () -> {
```

Comment:
You don't need to load the clusterAccount, you can just use the clusterId as the lock key.  This way you avoid another DB call at this point

Reply:
@ibuildthecloud I'll change the lock definition, but looks like we still need a clusterAccount object for pool manager call below.